### PR TITLE
cp-300: Fix sidebar redirect

### PR DESCRIPTION
**Pull request by HIMDanny**
- Added query string enums
- Added a helper to add a query string to the URL
- Updated `useSidebarShow` hook
- Removed query strings in chat component links

### CODE REVIEW

File: frontend/src/libs/components/navigation-menu-wrapper/libs/constants.ts [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+});
+
 const SIDEBAR_ROUTES: Route[] = [
   { path: AppRoute.ROOT, name: 'home', icon: 'home' },
-  { path: AppRoute.MEDITATION, name: 'meditation', icon: 'meditation' },
-  { path: AppRoute.CHATS, name: 'chats', icon: 'chats' },
-  { path: AppRoute.JOURNAL, name: 'journal', icon: 'journal' },
-  { path: AppRoute.PROFILE, name: 'profile', icon: 'user' },
+  {
+    path: urlWithQueryString(AppRoute.MEDITATION),
```
Comment (v0ldemar01) [inline frontend/src/libs/components/navigation-menu-wrapper/libs/constants.ts]: Can't we handle it via callback of setSidebarMode?

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: frontend/src/libs/enums/query-strings/sidebar-mode-query-string-value.enum.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const SidebarModeQueryStringValue = {
```
Comment (liza-veis) [inline frontend/src/libs/enums/query-strings/sidebar-mode-query-string-value.enum.ts]:
```suggestion
const SidebarMode = {
```

File: frontend/src/libs/helpers/get-url-with-query-string.helper.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { type AppQueryStringKey, type AppRoute } from '#libs/enums/enums.js';
+import { type ValueOf } from '#libs/types/types.js';
+
+type QueryParameters = Record<ValueOf<typeof AppQueryStringKey>, string>;
+
+const getUrlWithQueryString = (
```
Comment (liza-veis) [inline frontend/src/libs/helpers/get-url-with-query-string.helper.ts:6]: should accept path and parameters and return url, not function

File: frontend/src/pages/chat/components/chat-sidebar/chat-sidebar.tsx [hunk 4]
```diff
@@ -72,10 +70,10 @@ const ChatSidebar: React.FC<Properties> = ({
         </div>
         <div className={styles['chat-list']}>
           {filteredElements.map((filteredChat) => {
-            const chatLink = (AppRoute.CHATS_$ID.replace(
+            const chatLink = AppRoute.CHATS_$ID.replace(
               ':id',
               String(filteredChat.id),
-            ) + '?sidebarMode=hide') as typeof AppRoute.CHATS_$ID;
+            ) as typeof AppRoute.CHATS_$ID;
```
Comment (liza-veis) [inline frontend/src/pages/chat/components/chat-sidebar/chat-sidebar.tsx]:
```suggestion
            ) as ValueOf<typeof AppRoute>;
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/libs/helpers/get-url-with-query-string.helper.ts (hunk 3 above)
Comment (HIMDanny) [reply frontend/src/libs/helpers/get-url-with-query-string.helper.ts:6]: oh, I see. Ok, will rewrite it

File: frontend/src/libs/components/navigation-menu-wrapper/libs/constants.ts (hunk 1 above)
Comment (HIMDanny) [reply frontend/src/libs/components/navigation-menu-wrapper/libs/constants.ts]: but setSidebarMode is used in useSidebarState, and we can use hooks only in components. I didn't get what you mean

Comment (HIMDanny) [reply frontend/src/libs/components/navigation-menu-wrapper/libs/constants.ts]: moreover, the logic of the hook is focused on button clicks. not on links

**Review by liza-veis — APPROVED** [review]

File: frontend/src/pages/chat/chat.tsx [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
 
 import { ChatLayout, ChatSidebar } from './components/components.js';
 import styles from './styles.module.scss';
 
 const Chat: React.FC = () => {
+  const navigate = useNavigate();
   const { isSidebarShown, setIsSidebarShown } = useSidebarState();
 
   const handleBackButtonPress = useCallback(() => {
+    navigate(AppRoute.CHATS);
```
Comment (v0ldemar01) [inline frontend/src/pages/chat/chat.tsx:18]: But we have appActions for such purpose

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/chat/chat.tsx (hunk 5 above)
Comment (HIMDanny) [reply frontend/src/pages/chat/chat.tsx:18]: unfortunately, it doesn't work properly like this. It doesn't add the query string after redirection with `appActions`

Comment (HIMDanny) [reply frontend/src/pages/chat/chat.tsx:18]: so the option is to use the navigation from react-router like this

Comment (v0ldemar01) [reply frontend/src/pages/chat/chat.tsx:18]: But there is no query string in `navigate(AppRoute.CHATS);`

Comment (v0ldemar01) [reply frontend/src/pages/chat/chat.tsx:18]: Then how does it work?

Comment (HIMDanny) [reply frontend/src/pages/chat/chat.tsx:18]: yep, but there is `setIsSidebarShown(true)` below

Comment (HIMDanny) [reply frontend/src/pages/chat/chat.tsx:18]: I guess navigate in appActions rewrites the URL after applying the query string

**Review by OlenaYatseniuk — APPROVED** [review]
Looks good!

File: frontend/src/pages/chat/chat.tsx (hunk 5 above)
Comment (v0ldemar01) [reply frontend/src/pages/chat/chat.tsx:18]: navigate in appActions has the same logic as yours navigate. Btw, I've checked locally and both solutions work equally

Comment (HIMDanny) [reply frontend/src/pages/chat/chat.tsx:18]: hm, that's strange, cause I've also tested. Did you write like this?
```ts
  const handleBackButtonPress = useCallback(() => {
    dispatch(appActions.navigate(AppRoute.CHATS));
    setIsSidebarShown(true);
  }, [setIsSidebarShown, dispatch]);
```

Comment (v0ldemar01) [reply frontend/src/pages/chat/chat.tsx:18]: Ok, let's proceed with your solution, as we have not enough time to re-check it

**Review by v0ldemar01 — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**